### ISSUE TITLE
[data grid] Sort buttons in column headers are of insufficient color contrast

### ISSUE THREAD

**Issue by hungmanhle** [mui/mui-x#20189]
### Steps to reproduce

Steps:
1. Open this link to live example: https://mui.com/x/react-data-grid/#community-version-free-forever
2. Use chrome dev tools to force one of the column headers into the hover state.
3. Use an extension to figure out color values and contrasts such as https://chromewebstore.google.com/detail/accessible-color-picker/bgfhbflmeekopanooidljpnmnljdihld to figure out the contrast ratio of the sort button and its background.

### Current behavior

When hovering the column header and not hovering `sort` button, the buttons color contrast is lower than 3.0. 

<img width="1414" height="573" alt="Image" src="https://github.com/user-attachments/assets/2f7d4687-c1d3-48d5-bd36-a4f4bc40f795" />

### Expected behavior

The color contrast of the buttons should be 3.0 or higher. 

### Context

An a11y audit flagged that these buttons do not have sufficient contrast to be considered accessible for those with impaired vision. 

See https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: datagrid, a11y, color contrast, sort, button

**Comment by arminmeh** [comment]
@mui/xgrid 
Should we remove the hover icon?

1. It is inconsistent with the sorted column, where you don't get any additional visual indicator on hover
2. It seems that none of the other popular grids show an icon on hover

**Comment by MBilalShafi** [comment]
> [@mui/xgrid](https://github.com/orgs/mui/teams/xgrid) Should we remove the hover icon?
> 
>     1. It is inconsistent with the sorted column, where you don't get any additional visual indicator on hover
> 
>     2. It seems that none of the other popular grids show an icon on hover

Having a hover icon does slightly help to visually guess next sort order when unsorted, but I am OK with removing it if it isn't a common pattern and also because we will still have a "pointer" cursor, which could be an decent identification too.

We'll have to remove [**columnUnsortedIcon**](https://github.com/mui/mui-x/blob/28f7e7747c3ca66c601af719a38935180440f5c6/packages/x-data-grid/src/models/gridIconSlotsComponent.ts#L43) slot though which could be a breaking change.

**Comment by arminmeh** [comment]
> We'll have to remove [columnUnsortedIcon](https://github.com/mui/mui-x/blob/28f7e7747c3ca66c601af719a38935180440f5c6/packages/x-data-grid/src/models/gridIconSlotsComponent.ts#L43) slot though which could be a breaking change.

We can keep the slot, but make it empty by default.
